src/dropdown/dropdown.js

```diff
@@ -187,17 +187,30 @@ angular.module('ui.bootstrap.dropdown', ['ui.bootstrap.position'])
     }
   };
 
+  function toggleDropdownUp() {
+    var clientRect = $element[0].getBoundingClientRect(),
+      dropdownHeight = scope.getDropdownElement().outerHeight();
+
+    if (clientRect.top > dropdownHeight &&
+        window.innerHeight - clientRect.top < clientRect.height + dropdownHeight) {
+      return true;
+    }
+    return false;
+  }
+
   scope.$watch('isOpen', function(isOpen, wasOpen) {
     if (appendTo && self.dropdownMenu) {
-      var pos = $position.positionElements($element, self.dropdownMenu, 'bottom-left', true),
-        css,
+      var pos = $position.positionElements($element, self.dropdownMenu, 'auto bottom-left', true),
         rightalign,
-        scrollbarWidth;
-
-      css = {
-        top: pos.top + 'px',
-        display: isOpen ? 'block' : 'none'
-      };
+        scrollbarWidth,
+        css = {
```

Comment:
Grouping the css setting of this object hurts readability - better to move it back to how it was prior.